pkg/controller/configuration/controller.go

```diff
@@ -89,6 +89,38 @@ func (r *Reconciler) Reconcile(id controller.ID) (controller.Result, error) {
 			log.Warnf("Failed to reconcile configuration %s, %s", configurationID, err)
 			return controller.Result{}, err
 		}
+
+		target, err := r.topo.Get(ctx, topoapi.ID(configurationID))
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return controller.Result{}, err
+			}
+			return controller.Result{}, nil
+		}
+
+		configurableAspect := &topoapi.Configurable{}
+		err = target.GetAspect(configurableAspect)
+		if err != nil {
+			return controller.Result{}, err
+		}
+
+		initConfig := &configapi.Configuration{
```

Comment:
I think we should create an initial configuration entry when the target is added otherwise there is no way to find the config in the transaction controller to update source config in transaction.

Reply:
Yeah this is definitely something I overlooked... but I think this should actually be done by the transaction controller if it does not find a `Configuration` when applying a `Transaction`.